paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt

```diff
@@ -79,139 +80,139 @@ class PaparazziPlugin : Plugin<Project> {
-      // https://android.googlesource.com/platform/tools/base/+/96015063acd3455a76cdf1cc71b23b0828c0907f/build-system/gradle-core/src/main/java/com/android/build/gradle/tasks/MergeResources.kt#875
-
-      val moduleResourceDirs = variant.runtimeConfiguration
-        .artifactsFor(ArtifactType.ANDROID_RES.type) { it is ProjectComponentIdentifier }
-        .artifactFiles
-
-      val aarExplodedDirs = variant.runtimeConfiguration
-        .artifactsFor(ArtifactType.ANDROID_RES.type) { it !is ProjectComponentIdentifier }
-        .artifactFiles
-
-      val packageAwareArtifactFiles = variant.runtimeConfiguration
-        .artifactsFor(ArtifactType.SYMBOL_LIST_WITH_PACKAGE_NAME.type)
-        .artifactFiles
-
-      val writeResourcesTask = project.tasks.register(
-        "preparePaparazzi${variantSlug}Resources",
-        PrepareResourcesTask::class.java
-      ) { task ->
-        val android = project.extensions.getByType(BaseExtension::class.java)
-        val nonTransitiveRClassEnabled =
-          (project.findProperty("android.nonTransitiveRClass") as? String)?.toBoolean() ?: true
-
-        task.packageName.set(android.packageName())
-        task.artifactFiles.from(packageAwareArtifactFiles)
-        task.nonTransitiveRClassEnabled.set(nonTransitiveRClassEnabled)
-        task.mergeResourcesOutputDir.set(buildDirectory.asRelativePathString(mergeResourcesOutputDir))
-        task.targetSdkVersion.set(android.targetSdkVersion())
-        task.compileSdkVersion.set(android.compileSdkVersion())
-        task.mergeAssetsOutputDir.set(buildDirectory.asRelativePathString(mergeAssetsOutputDir))
-        task.projectResourceDirs.from(localResourceDirs)
-        task.moduleResourceDirs.from(moduleResourceDirs)
-        task.aarExplodedDirs.from(aarExplodedDirs)
-        task.paparazziResources.set(buildDirectory.file("intermediates/paparazzi/${variant.name}/resources.txt"))
-      }
-
-      val testVariantSlug = testVariant.name.capitalize(Locale.US)
+    val androidComponents = project.extensions.getByType(AndroidComponentsExtension::class.java)
+    androidComponents.onVariants { variant ->
+      project.afterEvaluate {
+        val variantSlug = variant.name.capitalize(Locale.US)
+        val testVariant = variant.unitTest ?: return@afterEvaluate
+
+        val internalArtifacts = variant.artifacts as ArtifactsImpl
+        val mergeResourcesOutputDir = internalArtifacts.get(InternalArtifactType.MERGED_RES)
+        val mergeAssetsOutputDir = variant.artifacts.get(SingleArtifact.ASSETS)
+        val projectDirectory = project.layout.projectDirectory
+        val buildDirectory = project.layout.buildDirectory
+        val gradleUserHomeDir = project.gradle.gradleUserHomeDir
+        val reportOutputDir = buildDirectory.dir("reports/paparazzi")
+        val snapshotOutputDir = project.layout.projectDirectory.dir("src/test/snapshots")
+
+        val localResourceDirs = project
+          .files(variant.sources.res!!.all.map { it.reversed() })
```

Comment:
Need to reverse the project res directory order compared to previous. Previously the list is in [ascending](https://cs.android.com/android-studio/platform/tools/base/+/mirror-goog-studio-main:build-system/gradle-core/src/main/java/com/android/build/gradle/api/BaseVariant.java;drc=5ac687029454dec1e7bd50697dabbc24d5a9943c;l=130) order, but in new api, it is in [descending](https://cs.android.com/android-studio/platform/tools/base/+/mirror-goog-studio-main:build-system/gradle-api/src/main/java/com/android/build/api/variant/SourceDirectories.kt;drc=5ac687029454dec1e7bd50697dabbc24d5a9943c;l=110) order.